### PR TITLE
WIP: Major revamp: index.tab, versions, error handling et al

### PR DESCRIPTION
**Pull request by shadowspawn**
Adding a Pull Request so some visibility on work in progress.

Pulling in much of the implementation from nvh, while (mostly) retaining the n commands and options.

- use index.tab instead of screen scraping for determining available versions
- support for full version, partial version, code name, and download builds (like nightly)
- consistent lookup of versions across all commands
- improved display of download errors
- et al

### CODE REVIEW

**Comment by shadowspawn** [comment]
Prerelease available via npm i n@next for v6. Release notes: https://github.com/tj/n/releases/tag/6.0.0-0

File: README.md [hunk 1]
```diff
@@ -104,68 +122,52 @@ wish to use node and npm, or are switching to a different way of managing them.
...
 
-    Options:
+List downloaded versions in cache:
 
-      -V, --version   Output version of n
-      -h, --help      Display help information
-      -q, --quiet     Disable curl output (if available)
-      -d, --download  Download only
-      -a, --arch      Override system architecture
+    nvh ls
```
Comment (DeeDeeG) [inline README.md]: `nvh ls` in the README.md is probably meant to be `n ls` for this repo.

(I tried `n ls` and it worked.)

File: README.md [hunk 2]
```diff
@@ -104,68 +122,52 @@ wish to use node and npm, or are switching to a different way of managing them.
...
 
-    Aliases:
+Display diagnostics to help resolve problems:
 
-      which   bin
-      use     as
-      list    ls
-      -       rm
-      stable  lts
+    nvh doctor
```
Comment (DeeDeeG) [inline README.md]: `nvh doctor` is probably meant to be `n doctor` for this repo.

(I did confirm running `n doctor` works on this branch.)

File: README.md (hunk 1 above)
Comment (shadowspawn) [reply README.md]: Fixed

File: README.md (hunk 2 above)
Comment (shadowspawn) [reply README.md]: Fixed

**Comment by shadowspawn** [comment]
Ha, I did a search for nvh close to the end of the work, but then added some more help and missed those. Thanks @DeeDeeG

**Pull request merged**